spec/breadcrumbs/breadcrumb_spec.rb

```diff
@@ -0,0 +1,90 @@
+# encoding: utf-8
+
+require 'spec_helper'
+
+require 'bugsnag/breadcrumbs/breadcrumb'
+
+RSpec.describe Bugsnag::Breadcrumbs::Breadcrumb do
+  describe "#name" do
+    it "is assigned in #initialize" do
+      breadcrumb = Bugsnag::Breadcrumbs::Breadcrumb.new("my message", nil, nil, nil)
+
+      expect(breadcrumb.name).to eq("my message")
+    end
+  end
+
+  describe "#type" do
+    it "is assigned in #initialize" do
+      breadcrumb = Bugsnag::Breadcrumbs::Breadcrumb.new(nil, "test type", nil, nil)
```

Comment:
its weird that you can set this to an invalid type to me.

Reply:
As this may be overwritten by callbacks anyway, it seems better to check this in a single place (the validator) than in the breadcrumb.

Reply:
in isolation i would have the setter validate. Its complicated by the metadata validation where that isnt possible however.